metals/src/main/scala/scala/meta/metals/sbtserver/SbtServer.scala

```diff
@@ -30,19 +33,29 @@ import org.scalasbt.ipcsocket.UnixDomainSocket
  * @param client client that can send requests and notifications
  *               to the sbt server.
  * @param runningServer The running client listening for requests from the server.
- *               Use runningServer.cancel() to stop disconnect to this server.
  *               Use runningServer.onComplete to attach callbacks on
  *               disconnect.
  *
  */
 case class SbtServer(
     client: JsonRpcClient,
     runningServer: CancelableFuture[Unit]
-)
+) {
+  def disconnect(): Unit = runningServer.cancel()
+}
 
 object SbtServer extends LazyLogging {
   private def fail(message: String) = Task.now(Left(message))
 
+  def readVersion(cwd: AbsolutePath): Option[String] = {
+    val path = cwd.resolve("project").resolve("build.properties")
+    val input = Files.newInputStream(path.toNIO)
```

Comment:
What if the file does not exist?

Reply:
Fixed in 40b1da5. I got a bit confused, thinking that this case will get swallowed by `Try { props.load(...) }`. 

By the way, recent versions of sbt (even 0.13.17) create `build.properties` if it doesn't exist.